addons/base_import/models/base_import.py

```diff
@@ -937,6 +946,16 @@ def parse_preview(self, options, count=10):
                         matches[index] = match.split('/')
             elif options.get('has_headers'):
                 matches = self._get_mapping_suggestions(headers, header_types, fields_tree)
+                if self._context.get("lang"):
+                    guessed_lang, guessed_lang_term_ratio = self.env['ir.translation']._guess_language_of_terms(
+                        terms=[re.sub(COLUMN_FIELD_HINT_REGEX, '', header)
+                               for header_full in headers for header in header_full.split('/')],
+                        types={'model'})
+                    if guessed_lang_term_ratio > 0.5 and self._context.get("lang") != guessed_lang:
+                        code_to_name = dict(self.env['res.lang'].get_installed())
+                        if guessed_lang in code_to_name:
+                            lang_warn = dict(lang=guessed_lang, lang_name=code_to_name[guessed_lang])
```

Comment:
Would use lang_code and lang_name to ease understanding of variable purpose

Reply:
Yes I think also. I was using lang because it was the name of the field that holds the lang code. Changed.